addons/road-generator/plugin.gd

```diff
@@ -1219,11 +1280,52 @@ func _create_roadpoint_pressed() -> void:
 		return
 
 	undo_redo.create_action("Add RoadPoint")
-	undo_redo.add_do_method(self, "_create_2x2_road_do", t_container, true)
-	undo_redo.add_undo_method(self, "_create_2x2_road_undo", t_container, true)
+	undo_redo.add_do_method(self, "_create_roadpoint_do", t_container)
+	undo_redo.add_undo_method(self, "_create_roadpoint_undo", t_container)
 	undo_redo.commit_action()
 
 
+## Add a RoadPoint to an existing RoadPoint
+func _create_roadpoint_do(t_container: RoadContainer):
+	var default_name = "RP_001"
+
+	if not is_instance_valid(t_container) or not t_container is RoadContainer:
+		push_error("Invalid RoadContainer")
+		return
+
+	# Get selected RoadPoint.
+	t_container.setup_road_container()
+	var selected_node = get_selected_node()
+	var first_road_point: RoadPoint
+	var second_road_point: RoadPoint
+
+	if not selected_node is RoadPoint:
+		print_debug("Couldn't add RoadPoint. Try selecting a RoadPoint, first.")
+		return
+
+	first_road_point = selected_node
+	second_road_point = RoadPoint.new()
+	second_road_point.name = second_road_point.increment_name(default_name)
+	first_road_point.add_road_point(second_road_point, RoadPoint.PointInit.NEXT)
+	set_selection(second_road_point)
+
+	t_container.update_edges() # Since we updated a roadpoint name after adding.
+
+
+func _create_roadpoint_undo(t_container: RoadContainer):
+	# Make a likely bad assumption that the last child of the RoadContainer is
+	# the one to be undone, but this is likely quite flakey.
+	# TODO: Perform proper undo/redo support, ideally getting add_do_reference
+	# to work property (failed when attempted so far).
+	var initial_children = t_container.get_children()
+
+	# Each RoadPoint handles their own cleanup of connected RoadSegments.
+	for i in range (len(initial_children)-1, 0, -1):
+		if initial_children[i] is RoadPoint:
+			initial_children[i].queue_free()
```

Comment:
Don't have to address right now, but when it comes to undo/redo functions and freeing objects, generally have to ad special undo/redo references so they get added to the stack. 

My newfound experience is that managing undo/redo ends up being a lot easier actually be using individual property or method calls, instead of top-level functions that affect lots of thing So more likely we'd create a for loop in the calling function that then sets up add_do_property changes and add_do_method on individual node-level operations.

Again, not to worry about for now, we can revisit once we get the baseline functionality working.